core/app/helpers/spree/images_helper.rb

```diff
@@ -15,21 +15,22 @@ def spree_image_tag(image, options = {})
     end
 
     def spree_image_url(image, options = {})
+      url_helpers = respond_to?(:main_app) ? main_app : Rails.application.routes.url_helpers
```

Comment:
@damianlegawiec I kept the `main_app` here if it's available - otherwise in `Rails.application.routes.url_helpers` there are no default options like protocol/host/port and it breaks on dev on the web app